module3/Projects/PairProject.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: page
+title: Mod3 Pair Project
+---
+
+# Coffee Shop
+
+## Kick Off
+
+[Learning to Pair](https://docs.google.com/presentation/d/1iSdA31sgik4ejeB95R4t2kvmh1p0rANC7VAfjc2Wgmo/edit#slide=id.gc6f90357f_0_0)
+
+### Objectives
+* Practice what we have learned about MVC applications
+* Work with a partner
+    * DTR
+    * Collaborative Git
+
+This project will help you practice the MVC skills we have learned, and push you to grow your collaborative skills. Each section of the project builds on the previous section; so, you will need to [pair program](https://www.geeksforgeeks.org/pair-programming/) with your partner.
+
+Before starting the project, read through the requirements (ask an instructor if you need any clarification).  Break down a list of tasks that need to be accomplished for each section, and determine who will 'drive' and who will 'navigate' for that task. Try to have equal opportunities to drive and navigate.
+
+As you work, it will be helpful to review the [Git for Pairs Lesson](/module2/lessons/Week2/GitForPairs).
+
+## Create your MVC Application
+
+In order to get started with the project, one partner will need to create the initial project structure so that it can be pushed to GitHub for collaboration.
+
+1. Create a new MVC Application
+2. Install the packages you need to:
+    * use PostgreSQL as your database
+    * use snake_case naming conventions for your database tables
+3. Create your database context object
+4. Commit your work and push to GitHub
+5. Add your partner as a collaborator
+
+From here on, you should not work on the `main` branch directly. You and your partner will need to determine when to create new branches, and when to merge those branches into main.
+
+## CRUD for an Item
+
+Our coffeeshop is going to sell 'Items'.  
+
+We want to use migrations to create and update our database (instead of creating an updating tables directly in pgAdmin); this is often referred to as a code-first approach. 
+
+1. In the database, we should store the following information about an Item:
+    * Name
+    * Price in Cents
+2. The Coffeeshop application should satisfy the following user stories. Each user story should be tested.
+```
+As a User
+When I visit "/items"
+Then I see a list of all items in the database
+```
+
+```
+As a User
+When I visit "/items/5"
```

Comment:
I just remembered we had talked about doing the different path naming convention for this spicy mini projec.t I still like that idea! I'd be in favor of doing these routes and at the beginning when we say that the project is more spicy we also mention that the routes will have a different naming structure, and maybe include a link to something from .NET that uses this convention instead of more RESTful?

// GET: Movies
// GET: Movies/Details/5
// GET: Movies/Create
// POST: Movies/Create
// GET: Movies/Edit/5
// POST: Movies/Edit/5
// GET: Movies/Delete/5
// POST: Movies/Delete/5

Reply:
Oh!  yes, I forgot we were going to do this 🙃 I think I'll specify the routes for the pages with views, and let them create the routes for the post actions.